mobile/cmp/multiZoneGrid/MultiZoneGridModel.ts

```diff
@@ -0,0 +1,421 @@
+/*
+ * This file belongs to Hoist, an application development toolkit
+ * developed by Extremely Heavy Industries (www.xh.io | info@xh.io)
+ *
+ * Copyright © 2023 Extremely Heavy Industries Inc.
+ */
+import '@xh/hoist/mobile/register';
+import {HoistModel, LoadSpec, PlainObject, Some, managed, XH, Awaitable} from '@xh/hoist/core';
+import {br, fragment} from '@xh/hoist/cmp/layout';
+import {action, bindable, makeObservable, observable} from '@xh/hoist/mobx';
+import {StoreRecordOrId, StoreTransaction} from '@xh/hoist/data';
+import {
+    Column,
+    ColumnSpec,
+    Grid,
+    GridConfig,
+    GridModel,
+    GridSorterLike,
+    multiFieldRenderer
+} from '@xh/hoist/cmp/grid';
+import {Icon} from '@xh/hoist/icon';
+import {castArray, forOwn, isEmpty, isFinite, isPlainObject, isString} from 'lodash';
+import {ReactNode} from 'react';
+import {MultiZoneMapperConfig, MultiZoneMapperModel} from './impl/MultiZoneMapperModel';
+import {MultiZonePersistenceModel} from './impl/MultiZonePersistenceModel';
+import {MultiZoneGridModelPersistOptions, Zone, ZoneLimit, ZoneMapping} from './Types';
+
+export interface MultiZoneGridConfig extends GridConfig {
+    /**
+     * Available columns for this grid. Note that the actual display of
+     * the multi-zone columns is managed via `mappings` below.
+     */
+    columns: Array<ColumnSpec>;
+
+    /** Mappings of columns to zones. */
+    mappings: Record<Zone, Some<string | ZoneMapping>>;
+
+    /** Optional configurations for zone constraints. */
+    limits?: Partial<Record<Zone, ZoneLimit>>;
+
+    /** Optional configs to apply to left column */
```

Comment:
Should we specify that this is an escape hatch? As specified, setting many of these will break stuff pretty badly